lib/charms/vault_k8s/v0/vault_managers.py

```diff
@@ -1447,3 +1448,60 @@ def _get_s3_parameters(self) -> dict[str, str]:
             if isinstance(value, str):
                 s3_parameters[key] = value.strip()
         return s3_parameters
+
+
+class RaftManager:
+    """Encapsulates the business logic for managing the bootstrap of a Vault cluster in Raft mode."""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        workload: WorkloadBase,
+        service_name: str,
+        storage_path: str,
+    ):
+        self._juju_facade = JujuFacade(charm)
+        self._workload = workload
+        self._service_name = service_name
+        self._storage_path = storage_path
+
+    def bootstrap(self, node_id: str, address: str) -> None:
+        """Bootstrap a Vault cluster in Raft mode.
+
+        This method will bootstrap a Vault cluster in Raft mode if it has not
+        already been bootstrapped. If the cluster is already bootstrapped, this
+        method will do nothing.
+        """
+        if not self._juju_facade.is_leader:
+            logger.debug("Only leader unit can bootstrap a Vault cluster")
+            raise ManagerError("Only the leader unit can bootstrap a Vault cluster")
+        if not self._juju_facade.planned_units == 1:
+            raise ManagerError("Bootstrapping a Vault cluster requires exactly one unit")
```

Comment:
This is the best I could come up with for ensuring a single unit. If anyone has a better approach I'd love to hear it.